### PR TITLE
remove deprecated kubelet endpoint /metrics/resource/v1alpha1

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind cleanup
/kind api-change

**What this PR does / why we need it**:
This PR removed the deprecated `kubelet` endpoint `metrics/resource/v1alpha1` which fist deprecated in v1.18, so it can be safely removed in v1.20.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
kubelet's deprecated endpoint `metrics/resource/v1alpha1` has been removed, please adopt to `metrics/resource`.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/priority backlog

### CODE REVIEW

**Review by MHBauer — APPROVED** [review]
I only vaguely understand the ` pkg/kubelet/server/server.go ` as part of updating the test in #93860 

Definitely don't know any part of the prometheus files, but I imagine they're dead now if server.go was the only caller.

File: pkg/kubelet/server/server.go [hunk 1]
```diff
@@ -378,14 +376,6 @@ func (s *Server) InstallDefaultHandlers(enableCAdvisorJSONEndpoints bool) {
 		compbasemetrics.HandlerFor(r, compbasemetrics.HandlerOpts{ErrorHandling: compbasemetrics.ContinueOnError}),
 	)
 
-	// deprecated endpoint which will be removed in release 1.20.0+.
-	s.addMetricsBucketMatcher("metrics/resource/v1alpha1")
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/server/server.go:382]: what about line 346 of this file? Perhaps needs to be removed a well.

File: pkg/kubelet/server/server.go [hunk 2]
```diff
@@ -378,14 +376,6 @@ func (s *Server) InstallDefaultHandlers(enableCAdvisorJSONEndpoints bool) {
...
-	v1alpha1ResourceRegistry := compbasemetrics.NewKubeRegistry()
-	v1alpha1ResourceRegistry.CustomMustRegister(stats.NewPrometheusResourceMetricCollector(s.resourceAnalyzer, stats.Config()))
-	s.restfulCont.Handle(path.Join(resourceMetricsPath, v1alpha1.Version),
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/server/server.go:385]: as you are removing this line, the entire module "k8s.io/kubernetes/pkg/kubelet/apis/resourcemetrics/v1alpha1" is not needed any longer. Consider removing

Comment (SergeyKanzhelev) [reply pkg/kubelet/server/server.go:385]: Just looked over, it has this comment =): 

```
// TODO(RainbowMango): We don't need to maintain this package anymore.
//  This package will be remove in release 1.20.0+. More details please refer to https://github.com/kubernetes/kubernetes/pull/86282.
```

File: pkg/kubelet/server/server.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/kubelet/server/server.go:382]: Thanks, I missed it. Done now.

File: pkg/kubelet/server/server.go (hunk 2 above)
Comment (RainbowMango) [reply pkg/kubelet/server/server.go:385]: Done. The whole package has been removed now.

**Comment by RainbowMango** [comment]
Hi, just a reminder, I think this ready to go. @dashpole

**Comment by dashpole** [comment]
Thanks a bunch @RainbowMango.  Nothing as satisfying as ripping out unneeded code.
/lgtm
/approve

**Comment by RainbowMango** [comment]
Thanks all! @MHBauer @SergeyKanzhelev @dashpole 

@justaugustus I guess we can give this a v1.20 label as per [Reopening development for v1.20](https://docs.google.com/document/d/1yI7o7R2bOB7QtkHUFMHMb5XObiJmzIAnJPuxhdAX-CA/edit#heading=h.6l9d3wg3ktoq), because it's kind of cleanup.

/milestone v1.20

**Pull request merged**